### PR TITLE
feature: add support for groups and pipelines

### PR DESCRIPTION
**Pull request by Bogdanp**
This is a WIP based on the discussion in #29.

### CODE REVIEW

File: dramatiq/middleware/pipelines.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        # from broker -> pipelines -> messages -> broker.
+        from ..message import Message
+
+        if exception is not None:
+            return
+
+        message_data = message.options.get("pipe_target")
+        if message_data is not None:
+            next_message = Message(**message_data)
+            if not message.options.get("pipe_ignore", False):
```
Comment (ericgj) [inline dramatiq/middleware/pipelines.py]: Thanks for this option. In my situation with pipelines constructed by less technical users, and more or less stand-alone, "fire-the-missiles" tasks, chaining results is mostly not needed.

**Comment by aequitas** [comment]
I like the feature to specify different options per message in the pipeline. I think the 'message_with_options' was added in a later iteration (or I just missed it when reading the PR earlier)? 
We have a use case where the same tasks needs to be routed to different workers (due to availability of IPv6 on that worker or not), this feature allows for a nice solution for this.

File: docs/source/cookbook.rst [hunk 2]
```diff
@@ -44,6 +44,103 @@ time an actor fails, even if the message is going to be retried.
...
+
+  pipe.get_result(block=True, timeout=5_000)
+
+To get the intermediate results of each step in the pipeline, you can
+call |pipeline_get_results|::
+
+  for res in pipe.get_results(block=True):
+    ...
+
+
```
Comment (bendemaree) [inline docs/source/cookbook.rst]: WDYT about specifically covering exception handling? I tried it out with a pipeline and found it entirely intuitive (`raise` in a pipeline step and subsequent steps do not execute) but perhaps it's worth mentioning?

Comment (Bogdanp) [reply docs/source/cookbook.rst]: I think that's something we can cover in the future, if there's a need. Result storage only supports results, not exceptions right now so we'd have to expand that scope, too. Might not even be necessary with #30.

**Pull request merged**